src/map.rs

```diff
@@ -360,25 +378,36 @@ where
     where
         Q: Hash + Equivalent<K>,
     {
-        if let Some((_, found)) = self.find(key) {
-            Some(found)
-        } else {
+        if self.is_empty() {
```

Comment:
Why check for empty here?

Reply:
It's a small perf hack to avoid computing the hash when there's nothing to compare. The former `fn find` did the same.